src/tls.c

```diff
@@ -8621,7 +8701,7 @@ int TLSX_KeyShare_Establish(WOLFSSL *ssl, int* doHelloRetry)
 
         /* Check consistency now - extensions in any order. */
         if (!TLSX_SupportedGroups_Find(ssl, clientKSE->group))
-            return BAD_KEY_SHARE_DATA;
+            continue;
```

Comment:
Can you explain or add comment for changing `BAD_KEY_SHARE_DATA` -> `continue`?

Reply:
hmmm....well, I noticed a test against openssl that was not passing. OpenSSL was sending supported groups with the first one being X25519 and that was causing this to return `BAD_KEY_SHARE_DATA`; it wasn't looking to the next algorithm in the list.  This allows it to look to the next algo in the list. 

So, the answer above probably isn't what you're looking for.  You probably want to understand what it is about my other changes that triggered the `BAD_KEY_SHARE_DATA`.  I will do a bit more digging.

Reply:
Now that begs, the question, why was `X25519` in that list. I'll go dig some more.

Reply:
Cha-Ching! In `DoTls13ClientHello()` (where the server handles client hello message) we eventually call `TLSX_SupportedCurve_Parse()` which then calls `TLSX_UseSupportedCurve()`.  Before my change, it never actually checked for whether we supported the client's curve.  Now that we check, X25519 doesn't get put in the list. 

Wow....that was pretty convoluted.